broadinstitute/single_cell_portal_core
# Update jQuery UI and postcss-cssnext per security alerts (SCP-3893)

This fixes reported security vulnerabilities in [jQuery](https://github.com/broadinstitute/single_cell_portal_core/security/dependabot/yarn.lock/jquery-ui/open) and [`postcss-cssnext`](https://github.com/broadinstitute/single_cell_portal_core/security/dependabot/yarn.lock/color-string/open).  The latter is technically unused in production, but seems relatively straightforward to update -- unlike `@rails/webpacker` and `react-scripts@3.4.4`.

It upgrades jQuery one minor version, from 1.12 to [1.13](https://blog.jqueryui.com/2021/10/jquery-ui-1-13-0-released/).  It also migrates away from the [deprecated `postcss-cssnext` package](https://www.npmjs.com/package/postcss-cssnext) to [a slightly more modern approach for Webpacker 5](https://candland.net/2021/upgrading-rails-webpacker-tailwindcss/) (sans e.g. Tailwind).

To test:
* Update dependencies, `yarn install --force`
* Go to a Study
* Go to Settings tab
* Change date in datepicker
* Save, refresh
* Verify the date you entered persists
* Verify that visual styling seems unchanged
* Search 3 genes (e.g.  [in "Tuberculosis in human female lymph"](https://singlecell-staging.broadinstitute.org/single_cell/study/SCP282/tuberculosis-in-human-female-lymph?genes=ABL1%2CADCY5%2CAGPAT2#study-visualize))
* Click around in Morpheus dot plot, including its menu
* Verify that UI and widget look OK

This satisfies SCP-3893.
